fix gateway error and write signature file on error

nymo opened the pull request:
Changed error handling and added gulp-plumber to keep gulp watch still processing files when error occurs. Also keeps writing the signature file for all successfully compressed images and just skip all with errors.

fixes #7

nymo committed:
fix gateway error and write signature file on error

cspicuzza commented:
Thanks for attacking this issue so fast!

I'm running a test with about 120 pngs.  The 502 was returned but the entire process halted.  When a timeout occurs can there be re-attempts to upload the same file then continue the process?

nymo commented:
@cspicuzza 
Currently I did not implement a way to automatically re-attempt the upload. I run a nightly cron where I update my images so all missed pictures will be fetched again on the next day. But we can add this as an additional feature for the gateway problem.
About the problem with the process halt...I added gulp-plumber as a requirement to this plugin. You should configure your gulpfile to add the plumber. This will not halt the process when an error is raised in the plugin and continue with the next picture. I updated the readme but here is a quick example:

```
var gulp = require('gulp');
var plumber = require('gulp-plumber');
...

gulp.task('tinypng', function () {
    gulp.src('images/src/**/*.{png,jpg,jpeg}')
        .pipe(plumber())
        .pipe(...
```

cspicuzza commented:
Thanks for your reply and code example! 

I originally meant to say the file gets skipped and the process kept going.   I ended up adding a halt process for when errors occurs.  With our app it's better to stop the whole process if there's an error.

Tinypng is one of many tasks in our chain of events.  We have 1800 svgs being converted to png, optimized, made into sprites, then uploaded to s3 buckets.   We are hoping to keep this process 100% automated.

Is there a planned release for the re-attempted upload feature?

nymo commented:
Okay I understand looks like a different scenario. I will open a new ticket for the re-attempted upload feature. I can't make you any promisses but I will try to have a look at this in the next week. Any contributions are welcome :)
